src/groupeddataframe/splitapplycombine.jl

```diff
@@ -237,6 +239,8 @@ const F_ARGUMENT_RULES =
     If the first or last argument is a function `fun`, it is passed a [`SubDataFrame`](@ref)
     view for each group and can return any return value defined below.
     Note that this form is slower than `pair` or `args` due to type instability.
+
+    If grouped data frame has zero groups then no transformations are applied.
```

Comment:
```suggestion
    If `gd` has zero groups then no transformations are applied.
```

So this is an interesting corner case indeed. Maybe we'd better throw an error in future releases (not in this PR since it should be backported)? Code that works with a nonzero number of groups is likely to try to access new columns and therefore fail later. Or could we call the function on a fake group with empty vectors just to find out the types of expected columns, and create them? Throwing an error would allow implementing this later.